llvm/lib/Transforms/IPO/GlobalOpt.cpp

```diff
@@ -1338,6 +1338,7 @@ deleteIfDead(GlobalValue &GV,
     if (DeleteFnCallback)
       DeleteFnCallback(*F);
   }
+  ReplaceableMetadataImpl::SalvageDebugInfo(GV);
```

Comment:
Is this line necessary? I believe `GV.removeDeadConstantUsers();` should already call this if `GV` is dead, as part of `constantIsDead(..., /*RemoveDeadUsers=*/true)`.

Reply:
The loop responsible for removing dead users and potentially calling constantIsDead() does not execute in this case. It looks like both the iterators I and E are null indicating it does not count debug intrinsic as users.

Reply:
It might be more appropriate to put this into `removeDeadConstantUsers` then, after the loop - I think we'd want to perform the salvage step for any constant that dies with debug users regardless of the pass that does it. However, that might end up affecting other tests; if so it can easily be left for another patch.

Reply:
Yes, I can investigate more and follow up on a different patch if it's a viable approach.